### PR TITLE
New one-page website

### PR DESCRIPTION
**Pull request by paf31**
screen shot 2017-06-18 at 8 31 55 am

### CODE REVIEW

**Comment by hdgarrood** [comment]
Looks great! The one thing I would say is perhaps we should link people directly to the getting started guide rather than suggesting they run `npm install -g purescript pulp` as they probably won't know what to do next; at that point I can imagine someone running `purs --help` and then `purs repl`, having it not work because nothing is installed, which could be a bit frustrating.

**Comment by hdgarrood** [comment]
> we might want to talk about the project goals near the top, and answer the question why do we really use PureScript and not something else?

Oh yeah, very much agree with this.

**Comment by kRITZCREEK** [comment]
I think it looks great, but I have no taste for design at all :D I might be a bit biased here, but I think it would be nice to showcase some of the tooling and how we focus

**Comment by soupi** [comment]
I like it but I also have a few comments if you don't mind.

I'm not a designer at all, but there are a few things I'd add/change:

- A link to a "getting started" tutorial at the top (as harry suggested)
- An example of a web UI application written in purs, preferably something non-trivial (as i'm guessing most people that arrive to that page is going to look for that
- Links to non trivial stuff made with purs since we clearly have some of those! (such as [slamdata](https://slamdata.com), [insect](https://insect.sh), [cubbit](https://github.com/aratama/cubbit)) - make people think: "wow, you can do that with purs?"
- Bigger size for the use-cases and examples, have more space between items so it'll be easier to focus on one thing at a time. Add bold or bigger headers to capture the eye when skimming.
- "we might want to talk about the project goals near the top, and answer the question why do we really use PureScript and not something else?" -- sounds good
- What Christoph said!
- Maybe also a link to https://news.purescript.org ? (this thing is alive and has a community)
- Maybe also write which release version is the latest?
- Link to the roadmap - so people know that the project is being worked on and that you have a plan
- Link to the PureScript by Example book - (this project have good resources to learn from)
- Edit the landing page for the github documentation repo to be more guided rather then a braindump

Here are a few links to other languages pages that I kinda like and think might take some inspiration from (and be better than!):

- [Guile Scheme](https://www.gnu.org/software/guile/)
- [Eta](https://eta-lang.org)
- [Kotlin](http://kotlinlang.org/)

**Comment by hdgarrood** [comment]
I'm not really keen on adding the most recent release, it's easy enough to find out what that is by going to GitHub and it would be yet another thing to do during each new release - making new compiler releases is complicated enough as is.

**Comment by hdgarrood** [comment]
Also the documentation repo is aimed at people who already have some basic level of familiarity with the language. It's not meant to be introductory - we have PureScript by Example for that. I also am not sure having it be more guided is a good thing to aim for, since, at least right now, I think the idea is that people know what they want to read about before they go to it.

Maybe we shouldn't really be linking to it at all..? (But then where do we link to it? Hm.)

I also just noticed that the text is all really tiny on mobile.

**Comment by hdgarrood** [comment]
Oops, I still have more things to say. I'm also not sure linking to the roadmap is a good idea especially not if the intention is to demonstrate activity. Only a very small proportion of development activity happens there.

**Comment by hdgarrood** [comment]
Oh also can we point existing pages to their new locations in the docs repo so that existing links won't 404?

**Comment by paf31** [comment]
> A link to a "getting started" tutorial at the top (as harry suggested)

👍 

> An example of a web UI application written in purs, preferably something non-trivial (as i'm guessing most people that arrive to that page is going to look for that

I'm not sure I'd like to highlight any particular application, since large PureScript codebases can be a little intimidating to new users to say the least. Instead, I'd like to just highlight short examples.

> Links to non trivial stuff made with purs since we clearly have some of those! (such as slamdata, insect, cubbit) - make people think: "wow, you can do that with purs?"

I think I'd prefer to put any lists of examples in the docs repo.

> "we might want to talk about the project goals near the top, and answer the question why do we really use PureScript and not something else?" -- sounds good

I'll make a separate comment where we can bikeshed the content for the top section.

> What Christoph said!

I would like to highlight tool support, since I think it's one of our strongest features. I think it should be highlighted in the top section (why do we use PureScript?)

> Maybe also a link to https://news.purescript.org ? (this thing is alive and has a community)

I think I would rather integrate such things into this page, if they can be done without a server component, and in a consistent way. I think we could also come back to this later.

> Maybe also write which release version is the latest?

I agree with Harry on this, I'd like to avoid adding content which will change regularly.

> Link to the PureScript by Example book - (this project have good resources to learn from)

Sounds like we have enough material for a "next steps" section below the Download/Try section, which can contain links to things like the book, the language guide and Pursuit.

> Edit the landing page for the github documentation repo to be more guided rather then a braindump

Looks like your PR took care of this already, thanks!

**Comment by paf31** [comment]
Here's a proposal for what to add:

- [ ] New section at the top, just below the header, which highlights the key reasons why we think PureScript is worth using:
  - Build real-world applications using functional techniques and expressive types
  - A large collection of libraries covering a variety of different types of development
  - Compile to readable JavaScript and reuse existing JavaScript code easily
  - Excellent tooling and editor support with instant rebuilds (can link to the docs repo here)
  - An active community with many learning resources
- [ ] Link to Getting Started near the top of the page
- [ ] Expand the _Learn More_ section with links to
  - Pursuit
  - The book
  - The docs repo

Thoughts? Is there anything else we want to highlight?

**Comment by gspia** [comment]
Hi, I'm a purescript newcomer and the new page definitely looks good! 

The _Learn more_ section could have links to some projects and maybe have some short intros of major libs with links to further examples. Maybe organized by topic: ui's, db's etc? 

"With the great libs you build your apps quickly..." + the four most intriguing topics organized like the "browser-server-apps-and more" -section? 

The _Get the compiler_ section could be extended a bit. Just after getting the compiler I think that people want to try it right away: was the installation ok and is the "hello world" app working etc. If there is no room for extra commands, maybe a link to the first example that shows the relevant commands?

The _Related projects_ link on documentation page under _Libraries_-section is 404. Since there is a link with the same name in under _Relaged Language_-section, is the _Related projects_ needed in _libraries_-section?

**Comment by paf31** [comment]
I had a go at a different style, incorporating some of the ideas above.

screen shot 2017-06-03 at 7 23 40 pm

**Comment by soupi** [comment]
I like it!

**Comment by paf31** [comment]
Another take, with the same content and a different layout:

screen shot 2017-06-04 at 9 36 36 am

**Comment by soupi** [comment]
I like the first one better tbh. I could also be nice to add more colors to the syntax highlight and also a third part for the example displaying the actual interactive UI that is being rendered if possible.

**Comment by paf31** [comment]
Yes, I thought about adding the actual UI in place of the compiled JS actually, but all three would work too. The only reason I included the JS was to show that the code is readable, but it's perhaps a bit noisy since this is a bit more than a Hello World example (I wonder if the mention of `Functor` might put some users off).

**Comment by soupi** [comment]
Yeah I agree that the JS might be a bit too noisy in this example. I think the most important things are the PureScript code, to see how nice it is, and the interactive UI, to see that it actually works and understand better what the code does. The JS can be left to something like try-purs I guess.

I really like the slogan btw! really nice.

**Comment by soupi** [comment]
Also maybe try to trim the text a bit or split it to more bullets and maybe bold some of the important bits? It feels a tiny bit like a wall of text and it might be worth to try to make it a bit more skimmable.

**Comment by paf31** [comment]
I was thinking to shorten it a little to

> Build real-world applications with strongly-typed functional programming!

**Comment by soupi** [comment]
I actually think the slogan in the pictures is perfect

**Comment by gspia** [comment]
One more idea / question: how much to highlight the similarities & differencies w.r.t. Haskell or is there room for these kind of things in the front page? I was first thinking that it might help raise interest on some seasoned players. The second thought was on this line was, how much to handle or highlight matters for people with different backgrounds: "if you are coming from javascript world, see this and that. Haskeller, please take a look of this2 and that2. Studying cs or math, why not give your career a boost with this3 and that3." 

And some of those items could stress out the why-question. In addition to the ones paf31 gives, there could be some reasons why purescript is to be preferred over javascript (e.g. the common reasons why the use of a statically typed functional language is considered to be a _good thing_, how it helps sw engineering task e.g.).

**Comment by paf31** [comment]
Here's my latest attempt:

screen shot 2017-06-06 at 8 31 18 pm

**Comment by kRITZCREEK** [comment]
I really like the last iteration! The example is not using the imported `log` though ;)

**Comment by paf31** [comment]
Note: we should document the minimum required version of Node for the NPM package.

**Comment by paf31** [comment]
Latest, with just some minor tweaks here and there:

screen shot 2017-06-07 at 7 48 18 pm

You can see it [here](rawgit.com/purescript/purescript.github.io/phil/one-page/index.html)

**Comment by paf31** [comment]
@garyb Any thoughts on the latest version?

**Comment by paf31** [comment]
Slight color changes after chatting with @garyb on Slack:

screen shot 2017-06-08 at 1 46 29 pm

**Comment by soupi** [comment]
Two suggestions:

1) bigger images for the bottom links to github and twitter
2) add a link to http://news.purescript.org ?

**Comment by paf31** [comment]
What about integrating PureScript News into the front page?

**Comment by rightfold** [comment]
Could be done, should be rewritten though

**Comment by paf31** [comment]
That's what I was thinking - it'd be nice to get it building on Travis so that I can just pull the build artifacts too. Maybe a project for later then.

I have a couple of minor suggestions for news.purescript.org as well, if we're planning to dust it off anyway.

**Comment by rightfold** [comment]
Most of the code is a mess. I want to use Hyper instead.

**Comment by paf31** [comment]
@hdgarrood Updated, thanks!

File: index.html [hunk 1]
```diff
@@ -156,7 +156,7 @@ <h2 class="text-center">A strongly-typed functional programming language that co
           <pre><span class="k">import</span> <span class="n">Prelude</span>
 <span class="k">import</span> <span class="n">Control.Monad.Eff.Console</span> (<span class="n">log</span>)
 
-<span class="n">greet</span> ∷ <span class="n">String</span> → <span class="n">String</span></span>
+<span class="n">greet</span> :: <span class="n">String</span> -> <span class="n">String</span></span>
 <span class="n">greet</span> name = <span class="s">"Hello, "</span> <> name <> <span class="s">"!"</span>
```
Comment (rightfold) [inline index.html]: Escape `<` as `&lt;`.

Comment (paf31) [inline index.html]: Good point, thanks!

**Comment by joneshf** [comment]
Who is the audience of this page? There look to be a couple of different groups of people we're intending to reach. The first five features feel targeted to someone that already knows a ton about typed programming, yet the "Get the compiler" section feels targeted to someone that knows the JS ecosystem well.

I think the change to have the list of benefits the language gives is great.

> * Build real-world applications using functional techniques and expressive types

This seems good.

> * A large collection of libraries covering a variety of different types of development

It feels like there's a thought here that is trying to be conveyed, but requires the reader to complete it. I'm not sure what it is though.

> * Compile to readable JavaScript and reuse existing JavaScript code easily

This seems like an excellent benefit!

> * Excellent tooling and editor support with instant rebuilds (can link to the docs repo here)

It feels like this purpose of this point is to quell concerns about using types and a compiler making things slow. Maybe that could be made more explicit? Maybe it's clear enough?

> * An active community with many learning resources

I think there's a point that could be punctuated by this. Maybe something about being supported while learning?

**Comment by paf31** [comment]
How about we move the top five points into a sublist called something like "An advanced type system", and down to the bottom of the list?

**Comment by joneshf** [comment]
I like that. Your revised phrasing seems better though. The revised phrasing is a list of benefits, rather than features. How would you feel about combining the two changes?

* Compile to readable JavaScript and reuse existing JavaScript code easily
* A large collection of libraries covering a variety of different types of development
* Excellent tooling and editor support with instant rebuilds
* An active community with many learning resources
* Build real-world applications using functional techniques and expressive types, such as
    * Algebraic data types and pattern matching
    * Row polymorphism and extensible records
    * Higher kinded types
    * Type classes with functional dependencies
    * Higher-rank polymorphism

**Comment by joneshf** [comment]
Particularly, I think "Compile to readable JavaScript and reuse existing JavaScript code easily" is wholly undersold. It's something that impresses people all the time.

**Comment by paf31** [comment]
That sounds good to me!

**Comment by paf31** [comment]
Updated!

**Comment by paf31** [comment]
I'd like to merge this, this week. So if anyone has any more comments, please speak up 😄

**Comment by garyb** [comment]
- I agree with Hardy's point about "A large collection of libraries covering a variety of different types of development" seeming like it's missing something. Maybe just be concrete? "A large collection of libraries for browser, server, and native development" or something if that's the intended implication. Or maybe going vaguer works too - "A large collection of libraries for diverse applications". I dunno, that's weird too.
- I'd flip the "Try PureScript" and "Quick start guide" buttons around, since "Try PureScript" could link to the very code sample above it in that case.
- "such as" should probably have a colon after it, before the sub-listing.
- I think that the previous code sample styling looked better, as the blue-background version draws the eye too much. Unless that really is the first thing you want people to look at on the page, but even then I think it could be toned down a bit.

I'd like to do some design tweaks around spacing and such, and probably the style of those buttons too, but it doesn't need holding up for that. :smile:

**Comment by paf31** [comment]
@garyb I just inverted the colors, I don't think it looks bad, what do you think?

screen shot 2017-06-19 at 3 28 37 pm

**Comment by soupi** [comment]
Maybe use the same color scheme as the examples at purescript.org?

**Comment by paf31** [comment]
Yeah that works:

screen shot 2017-06-21 at 9 02 15 am

**Comment by garyb** [comment]
Looks good 👍

**Comment by paf31** [comment]
I went with 

> An extensive collection of libraries for development of web applications, web servers, apps and more

I'll merge this and we can always change that line later.

Thanks for the help everyone!

**Pull request merged**